lib/internal/test_runner/reporter/spec.js

```diff
@@ -74,31 +100,18 @@ class SpecReporter extends Transform {
       case 'test:coverage':
         return getCoverageReport(indent(data.nesting), data.summary,
                                  reporterUnicodeSymbolMap['test:coverage'], colors.blue, true);
+      case 'test:summary':
+        // We report only the root test summary
+        if (data.file === undefined) {
```

Comment:
I'm not sure if this is the best way to handle this, or if there should be logic in `runner.js` similar to what already exists for other events.

Reply:
Our documentation states:

> `file <string> | <undefined>` The path of the test file that generated the summary. If the summary corresponds to multiple files, this value is undefined.  

At the moment, if I'm not mistaken, we're not passing any details to indicate that this summary comes from the runner's root test, except for `file === undefined`

Reply:
btw I totally agree, it seems a very bad and fragile approach

Reply:
OK, I'm good with this then.